lighthouse-core/gather/driver.js

```diff
@@ -216,20 +217,37 @@ class Driver {
   }
 
   /**
-   * Evaluate an expression in the context of the current page.
+   * Evaluate an expression in the context of the current page. If useIsolation is true, the expression
+   * will be evaluated in a content script that has access to the page's DOM but whose JavaScript state
+   * is completely separate.
    * Returns a promise that resolves on the expression's value.
    * @param {string} expression
+   * @param {{useIsolation: boolean}=} options
    * @return {!Promise<*>}
    */
-  evaluateAsync(expression) {
+  evaluateAsync(expression, options = {}) {
+    const contextIdPromise = options.useIsolation ?
+        this._getOrCreateIsolatedContextId() :
+        Promise.resolve(undefined);
+    return contextIdPromise.then(contextId => this._evaluateInContext(expression, contextId));
+  }
+
+  /**
+   * Evaluate an expression in the given execution context; an undefined contextId implies the main
+   * page without isolation.
+   * @param {string} expression
+   * @param {number|undefined} contextId
```

Comment:
can be `{number=}` FWIW

Reply:
yeah just seemed more explicit since something is always passed in but doesn't matter to me